connect/runtime/src/test/java/org/apache/kafka/connect/integration/DeadLetterQueueIntegrationTest.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.integration;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.errors.RetriableException;
+import org.apache.kafka.connect.storage.StringConverter;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.util.clusters.EmbeddedConnectCluster;
+import org.apache.kafka.test.IntegrationTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Category(IntegrationTest.class)
+public class DeadLetterQueueIntegrationTest {
+
+    private static final Logger log = LoggerFactory.getLogger(DeadLetterQueueIntegrationTest.class);
+
+    private static final String DLQ_TOPIC = "my-connector-errors";
+    private static final int NUM_RECORDS_PRODUCED = 20;
+    private static final int EXPECTED_CORRECT_RECORDS = 18;
+    private static final int EXPECTED_INCORRECT_RECORDS = 2;
+    private static final int CONSUME_MAX_DURATION_MS = 5000;
+
+    private EmbeddedConnectCluster connect;
+
+    @Before
+    public void setup() throws IOException {
+        connect = new EmbeddedConnectCluster(ConnectIntegrationTest.class);
+        connect.start();
+    }
+
+    @After
+    public void close() {
+        connect.stop();
+    }
+
+    @Test
+    public void testTransformationErrorHandling() throws Exception {
+        // create test topic
+        connect.kafka().createTopic("test-topic");
+
+        // produce some strings into test topic
+        for (int i = 0; i < NUM_RECORDS_PRODUCED / 2; i++) {
```

Comment:
Comments on the `ConnectIntegrationTest` apply here too. Please make sure the approach in both classes is the same.